Tools/cases_generator/generators_common.py

```diff
@@ -490,12 +491,19 @@ def _emit_block(
                     if reachable:
                         reachable = if_reachable
                     self.out.emit(rbrace)
+                elif tkn.kind == "INSTRUCTION_SIZE":
+                    self._emit_instruction_size(tkn, inst)
                 else:
                     self.out.emit(tkn)
         except StackError as ex:
             raise analysis_error(ex.args[0], tkn) from None
         raise analysis_error("Expecting closing brace. Reached end of file", tkn)
 
+    def _emit_instruction_size(self, tkn: Token, inst: Instruction | None) -> None:
```

Comment:
This should be a `ReplacementFunctionType` and registered in the emitter as a replacement function. See for example `DECREF_INPUTS`